diff -Naur kdebase-runtime-4.1.70/phonon/kcm/globalconfig.cpp kdebase-runtime-4.1.70.pulseaudio/phonon/kcm/globalconfig.cpp
--- kdebase-runtime-4.1.70/phonon/kcm/globalconfig.cpp	2008-07-17 23:32:48.000000000 +0200
+++ kdebase-runtime-4.1.70.pulseaudio/phonon/kcm/globalconfig.cpp	2008-10-21 18:49:35.000000000 +0200
@@ -28,6 +28,7 @@
 #include "phononnamespace_p.h"
 
 #include <QtCore/QList>
+#include <QtCore/QHash>
 #include <QtCore/QVariant>
 
 QT_BEGIN_NAMESPACE
@@ -158,6 +159,15 @@
                     | (defaultList.isEmpty() ? 0 : FilterHardwareDevices)
                     );
         }
+        // make PulseAudio the global default (assume it is already default in this list)
+        if (!list.isEmpty()) {
+            int firstIndex = list.first();
+            if (backendIface->objectDescriptionProperties(Phonon::AudioOutputDeviceType,
+                  firstIndex)["name"].toString() == "PulseAudio") {
+                list.removeFirst();
+                defaultList.prepend(firstIndex);
+            }
+        }
         defaultList += list;
     }
 
@@ -212,6 +222,15 @@
                     | (defaultList.isEmpty() ? 0 : FilterHardwareDevices)
                   );
         }
+        // make PulseAudio the global default (assume it is already default in this list)
+        if (!list.isEmpty()) {
+            int firstIndex = list.first();
+            if (backendIface->objectDescriptionProperties(Phonon::AudioCaptureDeviceType,
+                  firstIndex)["name"].toString() == "PulseAudio") {
+                list.removeFirst();
+                defaultList.prepend(firstIndex);
+            }
+        }
         defaultList += list;
     }
 
